paasta_tools/utils.py

```diff
@@ -3335,32 +3335,46 @@ def get_deploy_groups(self) -> Collection[str]:
         return self.config_dict["deployments"].keys()
 
     def get_docker_image_for_deploy_group(self, deploy_group: str) -> str:
+        key = "docker_image"
         try:
-            return self.config_dict["deployments"][deploy_group]["docker_image"]
+            deploy_group_config = self.config_dict["deployments"][deploy_group]
         except KeyError:
             e = f"{self.service} not deployed to {deploy_group}. Has mark-for-deployment been run?"
             raise NoDeploymentsAvailable(e)
+        try:
+            return deploy_group_config[key]
+        except KeyError:
+            e = f"The configuration for service {self.service} in deploy group {deploy_group} does not contain '{key}' metadata."
+            raise KeyError(e)
```

Comment:
Oh, I was thinking something much more simple like just splitting `return self.config_dict["deployments"][deploy_group]["docker_image"]` to 
```
deploy_group_config = self.config_dict["deployments"][deploy_group]
return deploy_group_config["docker_image"]
```
so that at least the NoDeploymentsAvailable exception is raised at the right place, and we effectively return the exact same thing as before otherwise.

But, this is nice to catch the additional error with more yelpy information in the error message.